app/data_models/progress_store.py

```diff
@@ -20,34 +20,52 @@ def __iter__(self) -> Iterator[tuple[str]]:
 
 class ProgressStore:
     """
-    An object that stores and updates references to sections and blocks
+    An object that stores and updates references to sections and list items
     that have been started.
     """
 
     def __init__(
-        self, in_progress_sections: Optional[Iterable[ProgressDictType]] = None
+            self,
+            in_progress_sections_and_repeating_blocks: Iterable[ProgressDictType]
+                                                       | None = None,
     ) -> None:
         """
-        Instantiate a ProgressStore object that tracks the status of sections and its completed blocks
+        Instantiate a ProgressStore object that tracks the progress status of Sections & Repeating Sections,
+        and their completed blocks, as well as Repeating Blocks for List Items.
+            - Standard Sections are keyed by Section ID, and a None List Item ID
+            - Repeating Sections (dynamic Sections created for List Items that have been added to a List Collector)
```

Comment:
Non blocker, I don't want to be annoying here, but we don't add list items to list collector.
```suggestion
            - Repeating Sections (dynamic Sections created for List Items that have been added using a List Collector)
```

Reply:
Sorted 👍